traits/ctraits.c

```diff
@@ -4283,6 +4283,85 @@ _get_trait_comparison_mode_int(trait_object *trait, void *closure)
     return PyLong_FromLong(i_comparison_mode);
 }
 
+
+/*-----------------------------------------------------------------------------
+|  Gets the 'property' value fields of a CTrait instance:
++----------------------------------------------------------------------------*/
+
+static PyObject *
+_trait_get_property(trait_object *trait, PyObject *args)
+{
+    PyObject *result, *temp;
+    if (trait->flags & TRAIT_PROPERTY) {
+        result = PyTuple_New(3);
+        if (result != NULL) {
+            PyTuple_SET_ITEM(result, 0, temp = trait->delegate_name);
+            Py_INCREF(temp);
+            PyTuple_SET_ITEM(result, 1, temp = trait->delegate_prefix);
+            Py_INCREF(temp);
+            PyTuple_SET_ITEM(result, 2, temp = trait->py_validate);
+            Py_INCREF(temp);
+            return result;
+        }
+        return NULL;
+    }
+    else {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
+}
+
+int _get_callable_argument_count(PyObject *callable){
+
+    if(PyMethod_Check(callable)){
+
+        PyFunctionObject *function_object =
+        (PyFunctionObject*)PyMethod_GET_FUNCTION(callable);
+
+        if(function_object){
+
+            PyCodeObject *code_object =
+            (PyCodeObject*)function_object->func_code;
+
+            if(code_object){
+                 return code_object->co_argcount -1;
+            }
+        }
+    }
+
+    else if(PyFunction_Check(callable)){
+        PyCodeObject* code_object =
+        (PyCodeObject*)PyFunction_GET_CODE(callable);
+
+        if(code_object){
+             return code_object->co_argcount;
+        }
+    }
+
+    else if(PyCallable_Check(callable)){
```

Comment:
Cython methods & functions are not caught by the other cases. Needs to be handled separately.